static/app/views/replays/detail/userActionsNavigator.tsx

```diff
@@ -26,17 +26,42 @@ type Props = {
   event: EventTransaction;
 };
 
+type PanelItemCenterProps = {
+  isHovered?: boolean;
+  isSelected?: boolean;
+};
+
 function UserActionsNavigator({event, crumbs}: Props) {
-  const {setCurrentTime} = useReplayContext();
+  const {setCurrentTime, currentHoverTime} = useReplayContext();
   const [currentUserAction, setCurrentUserAction] = useState<Crumb>();
+  const [closestUserAction, setClosestUserAction] = useState<Crumb>();
+
+  const {startTimestamp} = event;
+  const userActionCrumbs = onlyUserActions(transformCrumbs(crumbs));
+
+  useEffect(() => {
+    if (!currentHoverTime || !userActionCrumbs || !startTimestamp) {
+      setClosestUserAction(undefined);
+      return;
+    }
+    // TODO: Find a better way to find the closest user action without using `reduce` method
```

Comment:
What's wrong with reduce?

Reply:
think nothing is wrong, seems to be working as expected, just put a comment due to the fact that maybe there is a better way I didn't know, because reduce will iterate over the whole array over and over.